### ISSUE TITLE
Conversion to BiocBook 

### ISSUE THREAD

**Issue by js2264** [microbiome/OMA#375]
Hi OMA authors, 

I had the opportunity to chat with some of you at the EuroBioc23 regarding [`BiocBook`](https://www.bioconductor.org/packages/release/bioc/html/BiocBook.html). 

`BiocBook`-based packages allow automated versioning and rendering of a fully-fledged online book, along with automated versioning and deployment of a Docker image. Because it is built as a R package and complies with Bioconductor building requirements, it can even be submitted as a package to Bioconductor and be added to the books section of Bioconductor to be built and rendered by the BBS (e.g. see built https://bioconductor.org/checkResults/devel/books-LATEST/ and rendered: https://bioconductor.org/books/3.18/BiocBookDemo/). 

Here are two examples of `BiocBook`-based books I have worked on: 

- https://github.com/js2264/OHCA
- https://github.com/js2264/R4MS (adapted from the original R for mass spectrometry book from Laurent Gatto: https://github.com/rformassspectrometry/book)

**What would you think about adapting OMA to BiocBook infrastructure?** I would be happy to start working on the conversion, if you are interested in adopting this format. Let me know what you think!

**Comment by TuomasBorman** [comment]
Hello!

OMA is still in beta version, but our plan is to finish it by next Bioc release. For example, we have to still go through the text and improve the quality of it.

We are indeed interested to publish the book in Bioc. We actually already thought that adaptation of BiocBook would be good solution, but we have postponed the adaptation to Spring, when we are happy with the material. I think it is best to do the conversion as a last step before submitting.

We are more than happy to get help with the conversion. How about if we discuss about this thing in more detail in Spring?

-Tuomas

**Comment by js2264** [comment]
Hi @TuomasBorman, thanks for your answer. Of course it's totally ok, I will come back to you around the beginning of March. I just want to highlight that the writing experience is very comparable between bookdown and biocbook repositories, the only major change is that files are located in `inst/pages` rather than in the root, and the `_bookdown.yml` is now `inst/assets/_book.yml`. So transitioning the writing should be very straightforward. 

The more complex aspect of `BiocBooks` is to make sure that all dependencies are officially available (i.e. from CRAN or Bioconductor). This implies that `Remotes` are not allowed in the `DESCRIPTION` for a Bioconductor submission. I thought you might want to have that in mind during the following months when you finish up the text, as any of the `Remotes` deps will have to be removed prior to submission. That being said, it looks like all of them are already on CRAN/Bioconductor (except for `miaTime`?), so it's not clear to me why they are listed in Remotes. 

I took the liberty to fork the OMA book and give a try at switching to a BiocBook framework. It looks like all the dependencies are correctly being built, but with some fixes that remain. I'll work on it in the coming days (I have currently a bit of spare time) but I won't touch the page files, I hope that's ok!

**Comment by TuomasBorman** [comment]
Ok, cool, that's totally ok, thanks!

miaTime will be submitted to Bioc also at some point. As this has been a quite long project, we have to also prune and tidy-up little bit things (like this dependency-thing). Good to know that Remotes are not allowed, thanks! (This versioning is also something that we like -- the version of book matches with current Bioc)

**Comment by antagomir** [comment]
Yes very good. `BiocBooks` has many useful features also in terms of layout.

Does it support Quarto?

**Comment by TuomasBorman** [comment]
Yes, it supports Quarto

**Comment by js2264** [comment]
Hi @antagomir @TuomasBorman, 
Yes, all the functionalities described for Quarto websites and books are supported! 
I took the liberty to move page files around and convert them to .qmd. The automated website (only on Bioc devel for now) is available here: [https://js2264.github.io/OMA/docs/devel/](https://js2264.github.io/OMA/docs/devel/). I have opened a PR to check how changed files would show up. Feel free to postpone the review of this PR until ready, I mostly wanted to have it open so my fork (devel branch) can be directly compared to yours.
